dartagnan/src/main/java/com/dat3m/dartagnan/parsers/program/visitors/boogie/PthreadsProcedures.java

```diff
@@ -15,12 +15,15 @@ public static boolean handlePthreadsFunctions(VisitorBoogie visitor, Call_cmdCon
             case "pthread_cond_wait":
             case "pthread_cond_signal":
             case "pthread_cond_broadcast":
-            case "pthread_exit":
             case "pthread_mutex_destroy":
                 // TODO: These are skipped for now
                 VisitorBoogie.logger.warn(
                         "Skipped call to {} because the function is not supported right now.", funcName);
                 return true;
+            case "pthread_exit":
+                final Expression retVal = (Expression) ctx.call_params().exprs().expr(0).accept(visitor);
+                visitor.addEvent(EventFactory.newFunctionReturn(retVal));
```

Comment:
is it true that a `pthread_exit` acts simply as a function return?

Reply:
In principle it is reversed: a return acts as a `pthread_exit` I believe. Not that it really matters: both terminate the thread and produce a value that can be read by `pthread_join`.